### PR TITLE
[Filter/TensorRT] Add a sub-plugin filter for TensorRT

### PR DESCRIPTION
**Pull request by again4you**
This PR newly adds an sub-plugin filter for TensorRT and its related test cases.

It is a new version of #2771 since the rank count rules and related things are updated.
So I newly submitted a PR after dropping the old one.

Signed-off-by: Sangjung Woo sangjung.woo@samsung.com

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### v2
* Rebase the latest code

### v1
* Use `tensor_filter_subplugin` class
* Handle `DESTROY_NOTIFY` event to clean up the used GPU memory
  * Related issue: https://github.com/nnstreamer/nnstreamer/issues/2787

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc [hunk 1]
```diff
@@ -0,0 +1,484 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer Tensor_Filter, TensorRT Module
+ * Copyright (C) 2020 Sangjung Woo <sangjung.woo@samsung.com>
+ */
+/**
+ * @file   tensor_filter_tensorrt.cc
+ * @date   23 Oct 2020
+ * @brief  Mediapipe module for tensor_filter gstreamer plugin
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc:9]: Huh?? mediapipe??

**Review by myungjoo — APPROVED** [review]
LGTM except for the Doxygen entry of tensor_filter_tensorrt.cc

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+using Severity = nvinfer1::ILogger::Severity;
+
+/** @brief a global object of ILogger */
+class Logger : public nvinfer1::ILogger
+{
+	void log(Severity severity, const char* msg) override
+	{
+    switch (severity) {
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc:39]: Still have indent problem

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+
+/**
+ * @brief Configure the instance of the tensorrt_subplugin.
+ * @param[in] prop property of tensor_filter instance
+ */
+void tensorrt_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  /* Set model path */
+  if (prop->num_models != 1 || !prop->model_files[0]) {
+    ml_loge ("TensorRT filter requires one UFF model file\n");
```
Comment (helloahn) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc:171]: Q) Is the new line(`\n`) required?

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc (hunk 2 above)
Comment (again4you) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc:39]: Fixed. Thanks.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc (hunk 1 above)
Comment (again4you) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc]: Fixed.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc (hunk 3 above)
Comment (again4you) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorrt.cc]: Nice catch. Fixed.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request merged**